infrastructure/cdn-in-a-box/enroller/enroller.go

```diff
@@ -713,6 +713,125 @@ func enrollServerCapability(toSession *session, r io.Reader) error {
 	return err
 }
 
+// enrollFederation takes a json file and creates a Federation object using the TO API.
+// It also assigns a Delivery Service, the CDN in a Box admin user, IPv4 resolvers,
+// and IPv6 resolvers to that Federation.
+func enrollFederation(toSession *session, r io.Reader) error {
+	dec := json.NewDecoder(r)
+	var federation tc.AllDeliveryServiceFederationsMapping
+	err := dec.Decode(&federation)
+	if err != nil {
+		log.Infof("error decoding Server Capability: %s\n", err)
+		return err
+	}
+	mapping := federation.Mappings[0]
```

Comment:
This will segfault and crash the enroller if the federation has empty mappings

Reply:
Enrolling as many Federation mappings as are given in 286336f3f1.